whois-query/src/main/java/net/ripe/db/whois/query/pipeline/TermsAndConditionsHandler.java

```diff
@@ -15,6 +15,7 @@
     @Override
     public void channelActive(ChannelHandlerContext ctx) {
         ctx.channel().write(TERMS_AND_CONDITIONS);
+        ctx.pipeline().remove(this);
```

Comment:
Why is `remove(this)` done?

Reply:
Because the channel active event is fired when the client ip address is picked up from the proxy protocol header so the acl handlers can potentially close the connection based on the real ip. This causes the t&c to be displayed twice though so we remove this handler after it has been called once.